juju/juju
# Update Azure, AWS, LXD upstream SDK deps

Update go mod to pull in the latest upstream dependencies for Azure and AWS SDKs.
For Azure, a tweak was needed to extract the tenant ID from the auth header.

Also backport https://github.com/juju/juju/pull/15918 which tracks the Canonical LXD upstream. This is the first commit and forms the bulk of the changes. This backport resulted in an updated version of `go-humanize` so there's a few tweaks to some tabular output tests.

## QA steps

Azure:

```
juju bootstrap azure \
  --no-default-model \
  --storage-pool name=foo \
  --storage-pool type=azure \
  --storage-pool encrypted=true \
  --storage-pool vault-name-prefix=secret \
  --bootstrap-constraints="root-disk-source=foo"
  ```
Use Azure portal to check that encrypted disk is used for root disk..
```
juju deploy postgresql --storage pgdata=foo
juju status --storage
```
Use Azure portal to check that encrypted disk is used for postgres disk.

AWS:

```
juju bootstrap aws
juju deploy postgresql --storage pgdata=ebs
juju status --storage
```

LXD:

use QA steps in #15918

microk8s:
```
juju bootstrap microk8s
juju deploy snappass-test
```

## Links

**Jira card:** JUJU-5082
